rectools/metrics/debias.py

```diff
@@ -0,0 +1,206 @@
+
+    num_users_interacted_with_item = interactions_for_debiasing.groupby(Columns.Item, sort=False)[
+        Columns.User
+    ].nunique()
+
+    quantiles = num_users_interacted_with_item.quantile(q=[0.25, 0.75])
+    q1, q3 = quantiles.loc[0.25], quantiles.loc[0.75]
+    iqr = q3 - q1
+    max_border = int(q3 + config.iqr_coef * iqr)
+
+    item_outside_max_border = num_users_interacted_with_item[num_users_interacted_with_item > max_border].index
+
+    mask_outside_max_border = interactions_for_debiasing[Columns.Item].isin(item_outside_max_border)
+    interactions_result = interactions_for_debiasing[~mask_outside_max_border]
+    interactions_downsampling = interactions_for_debiasing[mask_outside_max_border]
+
+    interactions_downsampling = (
+        interactions_downsampling.sample(frac=1.0, random_state=config.random_state)
+        .groupby(Columns.Item)
+        .head(max_border)
+    )
+
+    result_dfs = [interactions_result, interactions_downsampling]
+    interactions_result = pd.concat(result_dfs, ignore_index=True)
+
+    return interactions_result
+
+
+def calc_debiased_fit_task(
+    metrics: tp.Iterable[DebiasableMetrikAtK],
+    interactions: pd.DataFrame,
+    debiasing_interactions_prev: tp.Optional[tp.Dict[DebiasConfig, pd.DataFrame]] = None,
+) -> tp.Dict[DebiasConfig, tp.Tuple[int, pd.DataFrame]]:
+    """
+    Calculate for each of the unique debias configs `k_max` and de-biasing `interactions`
+    to then apply them in the `fit` methods of the corresponding metrics.
+
+    Parameters
+    ----------
+    metrics : tp.Iteraple[DebiasableMetrikAtK]
+        Dict of metric objects to calculate, where key is metric name and value is metric object.
+    interactions : pd.DataFrame
+        Interactions or merging table with columns `Columns.User`, `Columns.Item`, `Columns.Rank` (for merging).
+        Obligatory only for some types of metrics.
+    debiasing_interactions_prev : dict(DebiasConfig->pd.DataFrame]), optinonal
+        De-biasing interactions for certain debias configs calculated earlier.
+
+    Returns
+    -------
+    dict(DebiasConfig->list[(int | pd.DataFrame)])
+        Dictionary, where key is debias config
+        and values are a tuple of the corresponding `k_max` and de-basing `interactions`.
+    """
+    debiasing_interactions = calc_debiased_different_configs(metrics, interactions, debiasing_interactions_prev)
+
+    max_k_for_config = {config: 0 for config in debiasing_interactions}
+    for metric in metrics:
+        max_k_for_config[metric.debias_config] = max(max_k_for_config[metric.debias_config], metric.k)
+
+    result = {
+        config: (max_k_for_config[config], d_interaciotns) for config, d_interaciotns in debiasing_interactions.items()
+    }
+    return result
+
+
+def calc_debiased_different_configs(
+    metrics: tp.Iterable[DebiasableMetrikAtK],
+    interactions: pd.DataFrame,
+    debiasing_interactions_prev: tp.Optional[tp.Dict[DebiasConfig, pd.DataFrame]] = None,
```

Comment:
Are you sure it's `debiasing`, not `debiased`? @blondered help

Reply:
`prev_debiased_interactions` here and everywhere